crates/stages/src/pipeline/mod.rs

```diff
@@ -270,10 +270,11 @@ where
                     Ok(unwind_output) => {
                         checkpoint = unwind_output.checkpoint;
                         self.metrics.stage_checkpoint(
-                            stage_id, checkpoint,
+                            stage_id,
+                            checkpoint,
                             // We assume it was set in the previous execute iteration, so it
                             // doesn't change when we unwind.
-                            None,
```

Comment:
Did I do this to right? Or is it a bug?

Reply:
oops yeah I didn't notice it. I think it'll mess up the `total` metric when we unwind, making `total = processed`. So we need to leave it as `Option<BlockNumber>` in metrics.

https://github.com/paradigmxyz/reth/pull/2975